src/git-repository.coffee

```diff
@@ -79,6 +79,9 @@ class GitRepository
     unless @repo?
       throw new Error("No Git repository found searching path: #{path}")
 
+    @watch = fs.watch path, { persistent: false, recursive: true }, (event, file) =>
```

Comment:
For some reason I totally skipped that part which mentioned the support for OS X only. That's unfortunate, but wouldn't be enough to build a directory tree and watch each directory? I don't think performance would degrade that much, but I haven't any benchmark to show so it's just a supposition :neutral_face: 

Alternatively there are other libraries which claim to solve the issue already: https://github.com/paulmillr/chokidar is one of them, and it has support for Windows, Linux and OS X (it seems to be well supported, but I haven't tried it yet).

@kevinsawicki: what do you think?